### DISCUSSION TITLE
Unicorn.call with arrays for select2 multiple

### DISCUSSION THREAD

**Discussion by vaughnd** [django-commons/django-unicorn#343] [Q&A] [answered]
Hi, I'm trying out the select2 example, but I want to use a 'multiple' field. The trouble is that it doesn't look like Unicorn.call handles arrays?

onchange="console.log($(this).val()); Unicorn.call('select2_country', 'select_country', $(this).val());"

Causes a 500 error ValueError: malformed node or string: <ast.Name object at 0x7f8945e6b0a0>

Not sure if I've just gone down the wrong path here. I couldn't find anything in the documentation around this.

**Comment by vaughnd** [comment, marked as answer]
For now I've worked around it with varargs in the Python method and using JS spread:
onchange="console.log($(this).val()); Unicorn.call('select2_country', 'select_country', ...$(this).val());"

Doesn't look like the Unicorn.call function supports anything more than basic string interpolation.


```
/**
 * Call an action on the specified component.
 */
export function call(componentNameOrKey, methodName, ...args) {
  const component = getComponent(componentNameOrKey);
  let argString = "";

  args.forEach((arg) => {
    if (typeof arg !== "undefined") {
      if (typeof arg === "string") {
        argString = `${argString}'${arg}', `;
      } else {
        argString = `${argString}${arg}, `;
      }
    }
  });

  if (argString) {
    argString = argString.slice(0, -2);
    methodName = `${methodName}(${argString})`;
  }

  component.callMethod(methodName, 0, null, (err) => {
    console.error(err);
  });
}
```

**Reply by adamghill** [reply]
It doesn't currently support anything other than strings, but it seems like a relatively straight-forward feature if you want to make a PR! 👍
